### PR TITLE
Updated Revit codes to align with node re-execution fixes in Dynamo

### PR DESCRIPTION
**Pull request by aparajit-pratap**
Refer PR: https://github.com/DynamoDS/Dynamo/pull/3523

### Reviewers ###
@Benglin 
@Randy-Ma

### CODE REVIEW

File: src/DynamoRevit/Models/RevitDynamoModel.cs [hunk 1]
```diff
@@ -426,9 +426,9 @@ private void ResetForNewDocument()
             foreach (var ws in Workspaces.OfType<HomeWorkspaceModel>())
             {
                 foreach (var node in ws.Nodes)
-                    node.ForceReExecuteOfNode = true;
+                    node.MarkNodeAsModified();
```
Comment (Benglin) [inline src/DynamoRevit/Models/RevitDynamoModel.cs]: I think resetting for a new Revit document we need all nodes to be force re-executed, no? In which case we need `MarkNodeAsModified(forceExecute: true)`. I'm just being afraid that removing that will cause the nodes to not re-execute.

**Comment by Benglin** [comment]
Other than that one point above, LGTM.

**Comment by Benglin** [comment]
Need another PR for R2015 too (which is to go in almost simultaneously).

File: src/DynamoRevit/Models/RevitDynamoModel.cs (hunk 1 above)
Comment (Randy-Ma) [inline src/DynamoRevit/Models/RevitDynamoModel.cs]: Is the function - MarkNodeAsModified - defined anywhere?

File: test/Libraries/RevitIntegrationTests/SelectionTests.cs [hunk 2]
```diff
@@ -84,7 +84,7 @@ public void SelectionDocModificationSync()
 
             TransactionManager.Instance.ForceCloseTransaction();
 
-            Assert.AreEqual(true, selectNode.ForceReExecuteOfNode);
+            Assert.AreEqual(true, selectNode.NeedsForceExecution);
```
Comment (Randy-Ma) [inline test/Libraries/RevitIntegrationTests/SelectionTests.cs]: Is NeedsForceExecution defined?

**Comment by Randy-Ma** [comment]
Except for the two points mentioned, LGTM!

**Comment by Randy-Ma** [comment]
@aparajit-pratap 
I think there may be one pending request which need to be merged before this one is merged. So be careful about this! Otherwise, it may cause a build error.

File: src/DynamoRevit/Models/RevitDynamoModel.cs (hunk 1 above)
Comment (Randy-Ma) [inline src/DynamoRevit/Models/RevitDynamoModel.cs]: agree

**Comment by Benglin** [comment]
You're right, @Randy-Ma, this pull request must only be merged after [its prerequisite](https://github.com/DynamoDS/Dynamo/pull/3523). Also, @aparajit-pratap you'll need to create another PR for Revit 2015 since that branch is exercised by the build machine. When all three pull requests are validated, they should be merged almost simultaneously, with 3253 going in first.

File: src/DynamoRevit/Models/RevitDynamoModel.cs (hunk 1 above)
Comment (aparajit-pratap) [inline src/DynamoRevit/Models/RevitDynamoModel.cs]: Prior to Steve's changes, the `RequiresRecalc` flag was set to `true`, which is why I let it be. However, I'll make the change as suggested. Don't see any harm in doing so.

**Pull request merged**